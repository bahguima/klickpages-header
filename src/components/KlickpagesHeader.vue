<template>
  <div class="navbar" role="navigation">
    <menu-switch />
    <div class="container no-padding">
      <home-link />
      <locale-dropdown />
      <nav-link-list />
      <mail-service-link />
      <notification-dropdown />
      <menu-user-dropdown />
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import MenuSwitch from './MenuSwitch.vue';
import HomeLink from './HomeLink.vue';
import LocaleDropdown from './LocaleDropdown.vue';
import NavLinkList from './NavLinkList.vue';
import MenuUserDropdown from './MenuUserDropdown.vue';
import MailServiceLink from './MailServiceLink.vue';
import NotificationDropdown from './notification/NotificationDropdown.vue';

export default {
  name: 'KlickpagesHeader',
  props: {
    klickartUrl: String,
  },
  components: {
    MenuSwitch,
    NavLinkList,
    LocaleDropdown,
    HomeLink,
    MenuUserDropdown,
    MailServiceLink,
    NotificationDropdown,
  },
  methods: mapActions({
    getTopBarConfig: 'topBar/getConfig',
  }),
  mounted() {
    this.getTopBarConfig(this.klickartUrl)
      .catch((error) => {
        console.error(error);
      });
  },
};
</script>
