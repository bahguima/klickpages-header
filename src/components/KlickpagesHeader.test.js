import { shallowMount, createLocalVue } from '@vue/test-utils';
import topBarRequest from '../services/klickartRequest/topBar';
import KlickpagesHeader from './KlickpagesHeader';

const mockHtml = 'mockedHTML';
jest.mock('../services/klickartRequest/topBar');

describe('components/KlickpagesHeader.vue', () => {
  const localVue = createLocalVue();
  let wrapper;

  describe('when component is mounted', () => {
    const klickartUrl = 'art.klickpages.com.br';
    const propsData = {
      klickartUrl,
    };

    beforeAll(() => {
      topBarRequest.mockImplementation(() => ({
        get: () => Promise.resolve({ data: mockHtml }),
      }));
      wrapper = shallowMount(KlickpagesHeader, { propsData, localVue });
    });

    it('should call topBarRequest with the klickartUrl', () => {
      expect(topBarRequest).toHaveBeenCalledWith({ klickartUrl });
    });

    it('should override the component html with the html returned by the request', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
