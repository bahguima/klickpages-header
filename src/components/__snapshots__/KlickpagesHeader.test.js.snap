// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/KlickpagesHeader.vue when component is mounted should override the component html with the html returned by the request 1`] = `
<div>
  mockedHTML
</div>
`;
